sdk/samples/iot/paho_iot_hub_pnp_component_sample.c

```diff
@@ -3,15 +3,18 @@
 
 #ifdef _MSC_VER
 // warning C4201: nonstandard extension used: nameless struct/union
-#pragma warning(push)
 #pragma warning(disable : 4201)
 #endif
 #include <paho-mqtt/MQTTClient.h>
 #ifdef _MSC_VER
-#pragma warning(pop)
+#pragma warning(default : 4201)
```

Comment:
I think @ahsonkhan was the one that suggested using push/pop for the pragma's?

Reply:
What's the rationale/benefit of **not** using push/pop? 

Push/pop let's you keep the warning state unmodified outside of this usage, which is nice. If you don't do that, we change the warning state to something other than what the outer-scope might have set it as.

Reply:
ah, ok. Am I understanding correctly that we with the push/pop, we shouldn't need to set the warning back to default, corrrect?  I will update this in all the samples.

Reply:
#ifdef _MSC_VER
#pragma warning(push)
// warning C4201: nonstandard extension used: nameless struct/union
#pragma warning(disable : 4201)
#endif
#include <paho-mqtt/MQTTClient.h>
#ifdef _MSC_VER
#pragma warning(pop)
#endif

Reply:
> ah, ok. Am I understanding correctly that we with the push/pop, we shouldn't need to set the warning back to default, corrrect?

yes

The `pop` happens near the end of the file or after the code block that is causing the warning you want to disable.